contributing/topics/defaults-and-breaking-changes.md

```diff
@@ -0,0 +1,128 @@
+# Defaults and Breaking Changes
+
+Over time, new and existing properities will need either a Default Value added or updated. This can be seen as a quick fix but could expose a breaking change that won't be caught by tests so doing a more in depth look into how Default values can impact Terraform is key.
+
+## Updating Default Values
+
+There are some cases where Azure updates the default value for an attribute when creating a new resource, and we would want to do the same for the provider but this is an easy breaking change to miss.
+
+We have a property like the following and Azure added a new spark version `3.4` and said that all new resources being created will be created with `3.4` as the default. 
+
+In Terraform, we start with:
+
+```hcl
+    "spark_version": {
+		Type:     pluginsdk.TypeString,
+		Optional: true,
+		Default:  "2.4",
+		ValidateFunc: validation.StringInSlice([]string{
+			"2.4",
+			"3.1",
+			"3.2",
+			"3.3",
+		}, false),
+	},
+```
+
+Then we would want to update `ValidateFunc` to include the new accepted value and update `Default` to `3.4` to keep it in line with Azure like so:
+
+```hcl
+    "spark_version": {
+		Type:     pluginsdk.TypeString,
+		Optional: true,
+		Default:  "3.4",
+		ValidateFunc: validation.StringInSlice([]string{
+			"2.4",
+			"3.1",
+			"3.2",
+			"3.3",
+			"3.4",
+		}, false),
+	},
+```
+
+But if we do that, people who have created that resource without the attribute specified will see a plan diff when upgrading to this version of the provider like so:
```

Comment:
I feel like there's supposed to be some example of the diff output here after `like so:`, but it looks like it was provided further down, maybe remove `like so:` here?
```suggestion
But if we do that, people who have created that resource without the attribute specified will see a plan diff when upgrading to this version of the provider.
```